framework/src/util/AnyMap.cpp

```diff
@@ -89,6 +89,65 @@ const Any& AtCompoundKey(const std::vector<Any>& v, const AnyMap::key_type& key)
     return v.at(index < 0 ? v.size() + index : index);
   }
 }
+Any AtCompoundKey(const std::vector<Any>& v,
+                  const AnyMap::key_type& key,
+                  Any&& defaultVal);
+
+Any AtCompoundKey(const AnyMap& m,
+                  const AnyMap::key_type& key,
+                  Any&& defaultVal)
+{
+  auto pos = key.find(".");
+  if (pos != AnyMap::key_type::npos) {
+    const auto head = key.substr(0, pos);
+    const auto tail = key.substr(pos + 1);
+    auto itr = m.find(head);
+    if (itr != m.end()) {
+      auto& h = itr->second;
+      if (h.Type() == typeid(AnyMap)) {
+        return AtCompoundKey(
+          ref_any_cast<AnyMap>(h), tail, std::move(defaultVal));
+      } else if (h.Type() == typeid(std::vector<Any>)) {
+        return AtCompoundKey(
+          ref_any_cast<std::vector<Any>>(h), tail, std::move(defaultVal));
+      }
+    }
+  } else {
+    auto itr = m.find(key);
+    if (itr != m.end()) {
+      return itr->second;
+    }
+  }
+  return defaultVal;
+}
+
+Any AtCompoundKey(const std::vector<Any>& v,
+                  const AnyMap::key_type& key,
+                  Any&& defaultval)
+{
+  auto pos = key.find(".");
+  const auto head = key.substr(0, pos);
+  const auto tail = (pos == AnyMap::key_type::npos) ? "" : key.substr(pos + 1);
+  int index = 0;
+  try {
+    index = std::stoi(head);
+  } catch (...) {
```

Comment:
I am not very fond of using `...` in general. E.g. on Windows with structured exception handling enabled, it will also swallow hardware exceptions (e.g. invalid memory accesses) and continuing the normal program flow is optimistic at best.